# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditSubActionHandler.cpp
# Commit: 69591833b1c2
# Full Hash: 69591833b1c2db9884926bdeadbe3183d17fb63f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-12-20 09:39:56
# Description:
#   Bug 1805554 - Make `HTMLEditor::MaybeSplitAncestorsForInsertWithTransaction` return `NS_ERROR_EDITOR_UNEXPECTED_DOM_TREE` if it got lost the editing host r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164771
# ==============================================================================

diff -r 857b1d139d11 -r 69591833b1c2 editor/libeditor/HTMLEditSubActionHandler.cpp
--- a/editor/libeditor/HTMLEditSubActionHandler.cpp	Tue Dec 20 04:38:47 2022 +0200
+++ b/editor/libeditor/HTMLEditSubActionHandler.cpp	Tue Dec 20 03:13:57 2022 +0000
@@ -4,6 +4,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "ErrorList.h"
 #include "HTMLEditor.h"
 #include "HTMLEditorInlines.h"
 #include "HTMLEditorNestedClasses.h"
@@ -8791,6 +8792,10 @@
     const Element& aEditingHost) {
   MOZ_ASSERT(IsEditActionDataAvailable());
 
+  if (NS_WARN_IF(!aEditingHost.IsInComposedDoc())) {
+    return Err(NS_ERROR_EDITOR_UNEXPECTED_DOM_TREE);
+  }
+
   if (NS_WARN_IF(!aStartOfDeepestRightNode.IsSet())) {
     return Err(NS_ERROR_INVALID_ARG);
   }
@@ -8824,7 +8829,10 @@
     }
   }
 
-  MOZ_DIAGNOSTIC_ASSERT(pointToInsert.IsSet());
+  // If we got lost the editing host, we can do nothing.
+  if (NS_WARN_IF(!pointToInsert.IsSet())) {
+    return Err(NS_ERROR_EDITOR_UNEXPECTED_DOM_TREE);
+  }
 
   // If the point itself can contain the tag, we don't need to split any
   // ancestor nodes.  In this case, we should return the given split point